prometheus-metrics-instrumentation-dropwizard/src/main/java/io/prometheus/metrics/instrumentation/dropwizard/DropwizardExports.java

```diff
@@ -0,0 +1,180 @@
+package io.prometheus.metrics.instrumentation.dropwizard;
+
+import io.dropwizard.metrics5.Timer;
+import io.dropwizard.metrics5.*;
+import io.prometheus.metrics.instrumentation.dropwizard.labels.CustomLabelMapper;
+import io.prometheus.metrics.model.registry.MultiCollector;
+import io.prometheus.metrics.model.snapshots.*;
+
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Collect Dropwizard metrics from a MetricRegistry.
+ */
+public class DropwizardExports implements MultiCollector {
+    private static final Logger LOGGER = Logger.getLogger(DropwizardExports.class.getName());
+    private MetricRegistry registry;
+    private MetricFilter metricFilter;
+    private Optional<CustomLabelMapper> labelMapper;
+
+    /**
+     * Creates a new DropwizardExports and {@link MetricFilter#ALL}.
+     *
+     * @param registry a metric registry to export in prometheus.
+     */
+    public DropwizardExports(MetricRegistry registry) {
+        super();
+        this.registry = registry;
+        this.metricFilter = MetricFilter.ALL;
+        this.labelMapper = Optional.empty();
+    }
+
+    /**
+     * Creates a new DropwizardExports with a custom {@link MetricFilter}.
+     *
+     * @param registry     a metric registry to export in prometheus.
+     * @param metricFilter a custom metric filter.
+     */
+    public DropwizardExports(MetricRegistry registry, MetricFilter metricFilter) {
+        this.registry = registry;
+        this.metricFilter = metricFilter;
+        this.labelMapper = Optional.empty();;
+    }
+
+    /**
+     * @param registry    a metric registry to export in prometheus.
+     * @param labelMapper a labelMapper to use to map labels.
+     */
+    public DropwizardExports(MetricRegistry registry, CustomLabelMapper labelMapper) {
+        this.registry = registry;
+        this.metricFilter = MetricFilter.ALL;
+        this.labelMapper = Optional.ofNullable(labelMapper);
+    }
```

Comment:
Thinking about it, maybe we should make the classes in the `labels` package not public (remove them from the public API and use them only internally), and make the label mapper configurable via the `DropwizardExports.Builder` instead. Something like this in `Builder`:

```java
        private static final ArrayList<MapperConfig> mapperConfigs = new ArrayList<>();

        /**
         * @param labels list of key/value pairs. Must be an even number.
         */
        public Builder mapperConfig(String match, String name, String... labels) {
            Map<String, String> labelsMap = new HashMap<>();
            for (int i=0; i*2<labels.length; i++) {
                labelsMap.put(labels[2*i], labels[2*i+1]);
            }
            mapperConfigs.add(new MapperConfig(match, name, labelsMap));
            return this;
        }
```

Then users could do something like this:

```java
        DropwizardExports.builder()
                .mapperConfig("test.dispatcher.*.*.*", "dispatcher_events_total", "action", "${1}", "outcome", "${2}_out")
                .register();
```

Just thinking out loud here, not sure if this is a good idea. But in general, the less classes are `public` the less backwards-compatible API we'll need to maintain. So what do you think?

Reply:
Added a `Builder` for `DropwizardExports`, this makes sense. However, if we want to make `labelMapper` via builder, it breaks the backward compatibility and ease of migration for people who are already using `client_java`. imho this could be a future release which makes it available via builder and let current users simply migrate by changing the import package names.